table/block.cc

```diff
@@ -762,7 +762,11 @@ BlockBasedTableOptions::DataBlockIndexType Block::IndexType() const {
   return index_type;
 }
 
-Block::~Block() { TEST_SYNC_POINT("Block::~Block"); }
+Block::~Block() {
+  // This sync point can be re-enabled once the right order for global static
+  // initialization/destruction across compilation units is determined.
```

Comment:
These are function-local statics and I don't believe the different compilation unit argument applies. That said, I'm ok with this PR as I don't know a proper fix.

Reply:
I should've explained more yesterday. We do not have a global static `SyncPoint`. It is local to the `SyncPoint::GetInstance` function, and it'll be initialized the first time that function is called. For such static variables, C++ does guarantee their destruction order is the reverse of their construction order; see here: http://eel.is/c++draft/basic.start.term#3.

Reply:
I misunderstood your first comment, and hence I went ahead with the commit. I should have clarified on this first, and I apologize for not doing so. 
Let me fix the comment in the Block destructor.

Reply:
#4380 for fixing the comment.